src/reactviews/pages/QueryResult/table/plugins/autoColumnSize.plugin.ts

```diff
@@ -0,0 +1,376 @@
+        rowElements.forEach((rowElement) => {
+            this.deleteRow(rowElement);
+        });
+        if (this._options.maxWidth) {
+            return widths.map((width) =>
+                Math.min(this._options.maxWidth!, width),
+            );
+        } else {
+            return widths.map((width) => width);
+        }
+    }
+
+    private getMaxColumnTextWidth(
+        columnDef: Slick.Column<T>,
+        colIndex: number,
+    ): number {
+        let texts: Array<string> = [];
+        let rowEl = this.createRow();
+        let data = this._grid.getData() as Slick.DataProvider<T>;
+        let viewPort = this._grid.getViewport();
+        let start = Math.max(0, viewPort.top);
+        let end = Math.min(data.getLength(), viewPort.bottom);
+        for (let i = start; i < end; i++) {
+            texts.push(data.getItem(i)[columnDef.field!]);
+        }
+        let template = this.getMaxTextTemplate(
+            texts,
+            columnDef,
+            colIndex,
+            data,
+            rowEl,
+        );
+        let width = this.getTemplateWidths([rowEl], [template])[0];
+        this.deleteRow(rowEl);
+        return width > this._options.maxWidth!
+            ? this._options.maxWidth!
+            : width;
+    }
+
+    private getTemplateWidths(
+        rowElements: JQuery[],
+        templates: (JQuery | HTMLElement | string)[],
+    ): number[] {
+        // Write all changes first then read all widths to prevent layout thrashing
+        // (https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing)
+        const cells: JQuery[] = templates.map((template, index) => {
+            let rowEl = rowElements[index];
+            let cell = jQuery(rowEl.find(".slick-cell"));
+            cell.append(template);
+            jQuery(cell).find("*").css("position", "relative");
+            return cell;
+        });
+
+        return cells.map((cell) => cell.outerWidth()! + 1);
+    }
+
+    private getMaxTextTemplates(
+        allTexts: string[][],
+        columnDefs: Slick.Column<T>[],
+        colIndices: number[],
+        data: Slick.DataProvider<T>,
+        rowElements: JQuery[],
+    ): (JQuery | HTMLElement | string)[] {
+        return columnDefs.map((columnDef, index) =>
+            this.getMaxTextTemplate(
+                allTexts[index],
+                columnDef,
+                colIndices[index],
+                data,
+                rowElements[index],
+            ),
+        );
+    }
+
+    private getMaxTextTemplate(
+        texts: string[],
+        columnDef: Slick.Column<T>,
+        colIndex: number,
+        data: Slick.DataProvider<T>,
+        rowEl: JQuery,
+    ): JQuery | HTMLElement | string {
+        let max = 0,
+            maxTemplate: JQuery | HTMLElement | string | undefined;
+        let formatFun = columnDef.formatter;
+        texts.forEach((text, index) => {
```

Comment:
this looks expensive at scale (e.g. lots of rows)

Reply:
@kburtram Did some perf testing on this, and since it just goes by the visible viewport, it only did the first 10 rows of each column for me (30 rows if I expand the grid to my full screen).  With 1024 columns (which seems like the upper recommended limit for # of columns), it took a couple seconds extra compared to if I had the setting off.

Reply:
Can you add a telemetry point with timing and number of columns here so that we can detect any egregious times to autosize?

Reply:
That's a good idea, I'll add that